### DISCUSSION TITLE
I can't animate the rotation of a vector shape from one end only

### DISCUSSION THREAD

**Discussion by AndrewPutter** [keiwando/toonsquid#580] [Q&A] [answered]
Hello there!

I'm trying to animate the rotation of a line-shaped vector by 360 degrees, using keyframes. By default, the pivot point for rotation is in the middle of the shape. I want the right-hand end of the line-like shape to sweep a 360 degree arc, but for the left-hand end to remain fixed, just like how the hand of a clock moves. I have tried moving the pivot point, but when I try to animate the rotation using keyframes, and play the resulting animation, it always animates the proportions of the entire shape, somehow animating the control points, squishing the line-like shape as it rotates.  

I've tried looking for a related discussion, but can't find one. My sincere apologies if there is one, and I've missed it!

Thanks so much!

https://github.com/keiwando/toonsquid/assets/25896555/e5b1f63c-e405-4828-863a-8c187dc9bde8

**Comment by keiwando** [comment, marked as answer]
Hi!
In the transform tool, tap on the path icon at the top to disable the "Edit path" mode. That's why it's currently editing the control points rather than the layer rotation.

<img width="222" alt="Screenshot 2024-05-14 at 17 01 52" src="https://github.com/keiwando/toonsquid/assets/12989570/f3ef5fe4-3e79-426d-a08e-d428acf93967">

**Reply by AndrewPutter** [reply]
Sadly that doesn't seem to work. 

When I disable the Edit Path mode by tapping on the path icon at the top while in the transform tool, I lose the ability to move the pivot point. This means that I can only rotate from the center of the object, and not the end that I want to rotate from. To get the editable pivot point back, the Path Mode icon at the top has to be blue.

I don't know what I'm doing wrong. : )

I really appreciate your help and would so appreciate being able to solve this. All the projects I currently want to make require rotating elements from one end, so I'm stuck until I can solve this problem.

I've attached another video, similar to the last, but showing how I lose the editable pivot point when I disable the Path Mode by tapping on the path icon when in the transform tool.

https://github.com/keiwando/toonsquid/assets/25896555/d7e8a6a9-51e1-4c54-80f5-31a0392e15ca

**Reply by Cat-Butler** [reply]
When you're moving the pivot point with Edit Path turned on, it's a temporary pivot for editing purposes (I want the line to be shorter, or curved in the middle, for the entire movie) rather than for keyframing/animation. 

With Edit Path turned off the pivot point might disappear at insufficient zoom levels, you will have to zoom in. Still can't see it? Zoom in some more. Repeat as needed. I don't know why it shows with editing on but not editing off. Keiwan, possible minor bug?

With something similar to your example at 1920x10180 on an iPad Mini 5, and a line that took up a decent part of the canvas width, I found I had to zoom in until I was only viewing a central segment of the line until the pivot point reappeared. Then it was move the pivot, pan the canvas, move the pivot, pan the canvas, until the pivot point was at the end of the line. Then lock (turn off) the pivot point editing and zoom out to animate and keyframe.

I suspect it might not have been so finicky with a larger (or at least higher resolution) screen and thus a larger working area for ToonSquid.

**Reply by AndrewPutter** [reply]
Cat-Butler, you're an angel - thank you!

That solved it.

Your suggestion almost didn't work! Then I discovered that if the shape (that I want to rotate from one end) is a vector path that has only an outline and no fill (which is how I've been drawing the line of my rotating watch-hand), I couldn't find a pivot point to edit. But when I made my watch-hand a long, narrow rectangle that had a visible fill, the pivot appeared in the fill, and could then be edited.

What a relief! A big thank you to you both, Keiwan and Cat-Butler. What an amazing app ToonSquid is! x

**Reply by keiwando** [reply]
Exactly, if the bounding box of the layer is too small or narrow on screen, the pivot handle gets hidden. If you zoom in far enough (or increase the line stroke width temporarily) it will show up. This also works for vector paths that only have a stroke and no fill.

And yes, the fact that the temporary pivot does not get hidden even if the bounding box is too small is indeed a bug. I have noted it down to fix that with a future update.
